plugin/trino-hive/src/test/java/io/trino/plugin/hive/BaseHiveConnectorTest.java

```diff
@@ -3907,6 +3926,74 @@ private void testMultipleWriters(Session session, HiveStorageFormat storageForma
         }
     }
 
+    @Test
+    public void testMultipleWritersWhenTaskScaleWritersIsEnabled()
```

Comment:
I'm not sure how valuable these tests are given that they would pass without scaling too.,

Reply:
Why would they pass without scaling?

Reply:
Due to `.isBetween(workers + 1, workers * taskMaxScaleWriterCount)`?

Isn't `workers + 1` a case without scaling? If it isn't, then how do you make sure your tests aren't flaky and scaling is guaranteed during test duration?

Reply:
> how do you make sure your tests aren't flaky and scaling is guaranteed during test duration?

It shouldn't happen because the amount of data (`sf1.orders`) is relatively high compared to writerMinSize which is 200KB. We have also decreased the parquet writer block size and orc stripe size. However, it can fail if the writer doesn't update the `physicalWrittenBytes` within the time period.

Reply:
I'm worried that this test might become flaky. Maybe you could run assertion in a loop until writers scale?